### PR TITLE
feat(inputs.lustre2): Skip brw_stats in case of insufficient permissions

### PR DESCRIPTION
**Pull request by lukeyeager**
Before this commit, the plugin entirely fails when run as anyone other than root, because brw_stats files are symlinks to /sys/kernel/debug/:
```
# readlink -f /proc/fs/lustre/osd-ldiskfs/MGS/brw_stats
/sys/kernel/debug/lustre/osd-ldiskfs/MGS/brw_stats
```
Which was causing errors like this:
```
2024-03-22T02:40:40Z E! [inputs.lustre2] Error in plugin: failed to read file /proc/fs/lustre/osd-ldiskfs/MGS/brw_stats: open /proc/fs/lus
tre/osd-ldiskfs/MGS/brw_stats: permission denied
```
After this change, the plugin continues to run and gather whatever metrics it can, even if brw_stats files are unreadable.

I'll rebase after https://github.com/influxdata/telegraf/pull/15042 is merged.

### CODE REVIEW

**Review by powersj — APPROVED** [review]

**Review by srebhan — APPROVED** [review]

**Comment by srebhan** [comment]
@lukeyeager thanks for your contribution. It would be nice if you can resolve the merge conflict...

File: plugins/inputs/lustre2/lustre2.go [hunk 1]
```diff
@@ -526,6 +527,10 @@ func (l *Lustre2) getLustreProcBrwStats(fileglob string, wantedFields []*mapping
 
 		wholeFile, err := os.ReadFile(file)
 		if err != nil {
+			if strings.Contains(err.Error(), "permission denied") {
```
Comment (srebhan) [inline plugins/inputs/lustre2/lustre2.go:530]: I just realized that
```suggestion
			if errors.Is(err, os.ErrPermission) {
```
is much better and recommended. Could you please change?

**Review by srebhan — COMMENTED** [review]
One more comment regarding error checking... It might be necessary to import `errors` if that's not done already...

File: plugins/inputs/lustre2/lustre2.go (hunk 1 above)
Comment (lukeyeager) [reply plugins/inputs/lustre2/lustre2.go]: I agree, that's better! Tested working:
```
[root@host tmp]# sudo -u telegraf ./telegraf -config ./telegraf.conf  --debug
2024-03-26T17:21:44Z I! Loading config: ./telegraf.conf
2024-03-26T17:21:44Z I! Starting Telegraf 1.31.0-cebe5c92 brought to you by InfluxData the makers of InfluxDB
2024-03-26T17:21:44Z I! Available plugins: 233 inputs, 9 aggregators, 31 processors, 24 parsers, 60 outputs, 6 secret-stores
2024-03-26T17:21:44Z I! Loaded inputs: cpu disk diskio kernel lustre2 mem processes swap system
2024-03-26T17:21:44Z I! Loaded aggregators:
2024-03-26T17:21:44Z I! Loaded processors:
2024-03-26T17:21:44Z I! Loaded secretstores:
2024-03-26T17:21:44Z I! Loaded outputs: prometheus_client
2024-03-26T17:21:44Z I! Tags enabled:
2024-03-26T17:21:44Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"", Flush Interval:10s
2024-03-26T17:21:44Z D! [agent] Initializing plugins
2024-03-26T17:21:44Z D! [agent] Connecting outputs
2024-03-26T17:21:44Z D! [agent] Attempting connection to [outputs.prometheus_client]
2024-03-26T17:21:44Z I! [outputs.prometheus_client] Listening on http://[::]:9273/metrics
2024-03-26T17:21:44Z D! [agent] Successfully connected to outputs.prometheus_client
2024-03-26T17:21:44Z D! [agent] Starting service inputs
2024-03-26T17:21:50Z D! [inputs.disk] [SystemPS] => unable to get disk usage ("/sys/kernel/debug/tracing"): permission denied
2024-03-26T17:21:50Z D! [inputs.lustre2] open /proc/fs/lustre/osd-ldiskfs/MGS/brw_stats: permission denied
2024-03-26T17:21:50Z D! [inputs.lustre2] open /proc/fs/lustre/osd-ldiskfs/fs1-MDT0000/brw_stats: permission denied
2024-03-26T17:21:50Z D! [inputs.lustre2] open /proc/fs/lustre/osd-ldiskfs/fs1-OST0000/brw_stats: permission denied
2024-03-26T17:21:50Z D! [inputs.lustre2] open /proc/fs/lustre/osd-ldiskfs/fs1-OST0001/brw_stats: permission denied
```

**Review by srebhan — APPROVED** [review]
Thanks a lot @lukeyeager!

**Pull request merged**